src/stratagus/construct.cpp

```diff
@@ -264,15 +258,10 @@ static int CclDefineConstruction(lua_State *l)
 					lua_pop(l, 1);
 				}
 				lua_pop(l, 1);
-				CConstructionFrame **cframe = &construction->Frames;
-				while (*cframe) {
-					cframe = &((*cframe)->Next);
-				}
-				(*cframe) = new CConstructionFrame;
-				(*cframe)->Percent = percent;
-				(*cframe)->File = file;
-				(*cframe)->Frame = frame;
-				(*cframe)->Next = nullptr;
+				auto& cframe = construction->Frames.emplace_back();
```

Comment:
Could the size be set before to avoid re-allocation?

Reply:
https://github.com/Wargus/stratagus/pull/637 created.